packages/devtools/lib/src/logging/logging.dart

```diff
@@ -705,3 +704,31 @@ class LogDetailsUI extends CoreElement {
     }
   }
 }
+
+class FrameInfo {
+  FrameInfo(this.number, this.elapsedMs, this.startTimeMs);
+
+  static const double kTargetMaxFrameTimeMs = 1000.0 / 60;
+
+  static FrameInfo from(Map<dynamic, dynamic> data) {
+    return FrameInfo(
+        data['number'], data['elapsed'] / 1000, data['startTime'] / 1000);
+  }
+
+  final int number;
+  final num elapsedMs;
+  final num startTimeMs;
+
+  bool frameGroupStart = false;
+
+  num get endTimeMs => startTimeMs + elapsedMs;
+
+  void calcFrameGroupStart(FrameInfo previousFrame) {
```

Comment:
nit: this method has a little bit of a code smell. probably cleaner to have a
frameGroupStart method and keep this logic out of this class.

Reply:
Moving this class to logging.dart was just a straight copy paste, but I'll look into it 👍

Reply:
nothing uses this logic anymore with the removal of the old bar chart. Removed method.